fe-code-review/code.js

```diff
@@ -0,0 +1,142 @@
+                    const collectorEmail = debtCollectors[i].email;
+                    const hashSum = crypto.createHash('sha256');
+                    const hashInput = `${idUser}-${idCollector}-${(new Date()).toISOString()}`;
+                    logDebug('hashInput=', hashInput);
+                    hashSum.update(hashInput);
+                    const requestKey = hashSum.digest('hex');
+                    logDebug('REQUEST KEY:', requestKey);
+
+                    const hash = Buffer.from(`${idUser}__${idCollector}`, 'utf8').toString('base64')
+
+                    if (!!(await db.setUserRequestKey(requestKey, idUser))
+                        && !!(await db.setUserCollectorRequestKey(requestKey, idUser, idCollector))) {
+
+                        /* prepare email */
+                        const sendConfig = {
+                            sender: config.projects[project].email.sender,
+                            replyTo: config.projects[project].email.replyTo,
+                            subject: 'Email subject',
+                            templateId: config.projects[project].email.template.collector,
+                            params: {
+                                downloadUrl: `https://url.go/download?requestKey=${requestKey}&hash=${hash}`,
+                                uploadUrl: `https://url.go/upload?requestKey=${requestKey}&hash=${hash}`,
+                                confirmUrl: `https://url.go/confirm?requestKey=${requestKey}&hash=${hash}`
+                            },
+                            tags: ['request'],
+                            to: [{ email: collectorEmail , name: collectorName }],
+                        };
+                        logDebug('Send config:', sendConfig);
+
+                        try {
+                            await db.setEmailLog({collectorEmail, idCollector, idUser, requestKey})
+                        } catch (e) {
+                            logDebug('extract() setEmailLog error=', e);
+                        }
+
+                        /* send email */
+                        const resp = await email.send(sendConfig, config.projects[project].email.apiKey);
+                        logDebug('extract() resp=', resp);
+
+                        // update DB with result
+                        await db.setUserCollectorRequestKeyRes(requestKey, idUser, idCollector, resp);
+
+                        if (!sentStatus[collectorName])
+                            sentStatus[collectorName] = {};
+                        sentStatus[collectorName][collectorEmail] = resp;
+
+                        if (!resp) {
+                            logError('extract() Sending email failed: ', resp);
+                        }
+                    }
+                }
+                await db.updateStatus(requestID, 100, '');
+
+                logDebug('FINAL SENT STATUS:');
+                console.dir(sentStatus, {depth: null});
```

Comment:
You mixing logger messages with console methods it increases the pollution in logging